mlir/lib/Dialect/QCO/Transforms/Decomposition/Helpers.cpp

```
@@ -0,0 +1,65 @@
+  // equivalent for this case
+  auto wrapped = remEuclid(angle + std::numbers::pi, 2 * std::numbers::pi) -
+                 std::numbers::pi;
+  if (std::abs(wrapped - std::numbers::pi) < angleZeroEpsilon) {
+    // Canonicalize the upper endpoint back to -pi so callers always receive a
+    // half-open interval [-pi, pi).
+    return -std::numbers::pi;
+  }
+  return wrapped;
+}
+
+std::size_t getComplexity(decomposition::GateKind type,
```

Cppcheck (2.20.0)

[style] The function 'getComplexity' is never used.

(unusedFunction)